builder/includes/associations.form.inc

```diff
@@ -298,3 +295,137 @@ function xml_form_builder_enable_association($form_name, $id) {
+      '#collapsible' => TRUE,
+      'content_model' => array(
+        '#type' => 'textfield',
+        '#title' => t('Content Model'),
+        '#required' => TRUE,
+        '#autocomplete_path' => ISLANDORA_CONTENT_MODELS_AUTOCOMPLETE,
+        '#description' => t('The content model to associate with a form. If the content model has no decendents it will not show up in autocomplete.'),
+        '#default_value' => $association['content_model'],
+      ),
+      'dsid' => array(
+        '#type' => 'textfield',
+        '#description' => t("The datastream ID where the object's metadata is stored."),
+        '#title' => t('Metadata Datastream ID'),
+        '#required' => TRUE,
+        '#default_value' => $association['dsid'],
+      ),
+      'form_name' => array(
+        '#type' => 'value',
+        '#title' => t('Form Name'),
+        '#value' => $form_name,
+      ),
+      'title_field' => array(
+        '#type' => 'select',
+        '#title' => t('Title Field'),
+        '#description' => t("The form field for the object's label."),
+        '#prefix' => '<div id="ahah-wrapper">',
+        '#suffix' => '</div>',
+        '#options' => $options,
+        '#default_value' => $selected,
+      ),
+      'id' => array(
+        '#type' => 'value',
+        '#value' => $id,
+      ),
+    ),
+  );
+  if (!variable_get('xml_form_builder_use_default_dc_xslts', FALSE)) {
+    $form['fieldset']['transform'] = array(
+      '#type' => 'select',
+      '#title' => t('XSL Transform'),
+      '#options' => xml_form_builder_get_transforms(),
+      '#default_value' => $association['transform'],
+      '#description' => t("An XSL transform for setting the Fedora object's Dublin Core metadata datastream."),
+    );
+    $form['fieldset']['self_transform'] = array(
+      '#type' => 'select',
+      '#title' => t('Self XSL Transform'),
+      '#options' => xml_form_builder_get_self_transforms(),
+      '#default_value' => $association['self_transform'],
+      '#description' => t('An optional transformation applied to form data prior to ingestion.'),
+    );
+  }
+  $form['fieldset']['file'] = array(
+    '#type' => 'file',
+    '#title' => t('Upload Template Document'),
+    '#description' => t('A sample metadata file used to prepopulate the form on ingest.'),
+  );
+  $form['fieldset']['submit'] = array(
+    '#type' => 'submit',
+    '#value' => t('Edit Association'),
+  );
+
+  return $form;
+}
+
+/**
+ * Default submit handler for the associations edit form.
+ *
+ * @param array $form
+ *   The drupal form.
+ * @param array $form_state
+ *   The drupal form state.
+ */
+function xml_form_builder_edit_association_form_submit(array $form, array &$form_state) {
+  module_load_include('inc', 'xml_form_builder', 'includes/associations');
+  $id = $form_state['values']['id'];
+  $form_name = $form_state['values']['form_name'];
+  $template = '';
+  $file_uploaded = $_FILES['files']['error']['file'] == 0;
```

Comment:
If we are using a file type of form element we should use [file_save_upload](https://api.drupal.org/api/drupal/includes!file.inc/function/file_save_upload/7.x) on it and use it from there. Alternatively could change the `#type`  to `managed_file` and use the fid instead.